### PR TITLE
[AMS-940] fix setting content type in file uploading

### PR DESCRIPTION
**Pull request by xakep139**

### CODE REVIEW

File: src/VStore/Sessions/SessionManagementService.cs [hunk 1]
```diff
@@ -52,6 +52,19 @@ public sealed class SessionManagementService
...
 
+        private static readonly IReadOnlyDictionary<string, string> ContentTypesMap =
+            new Dictionary<string, string>
+                {
+                    { FileFormat.Bmp.ToString().ToLowerInvariant(), "image/bmp" },
+                    { FileFormat.Gif.ToString().ToLowerInvariant(), "image/gif" },
+                    { FileFormat.Jpeg.ToString().ToLowerInvariant(), "image/jpeg" },
+                    { FileFormat.Jpg.ToString().ToLowerInvariant(), "image/jpeg" },
+                    { FileFormat.Png.ToString().ToLowerInvariant(), "image/png" },
+                    { FileFormat.Chm.ToString().ToLowerInvariant(), "application/x-chm" },
```
Comment (xakep139) [inline src/VStore/Sessions/SessionManagementService.cs]: This cuold be also `application/octet-stream`

File: src/VStore/Sessions/SessionManagementService.cs [hunk 2]
```diff
@@ -52,6 +52,19 @@ public sealed class SessionManagementService
...
+                    { FileFormat.Svg.ToString().ToLowerInvariant(), "image/svg+xml" },
```
Comment (xakep139) [inline src/VStore/Sessions/SessionManagementService.cs]: https://www.w3.org/TR/SVG/mimereg.html

File: src/VStore/Sessions/SessionManagementService.cs [hunk 3]
```diff
@@ -52,6 +52,19 @@ public sealed class SessionManagementService
...
+                    { FileFormat.Pdf.ToString().ToLowerInvariant(), "application/pdf" }
```
Comment (xakep139) [inline src/VStore/Sessions/SessionManagementService.cs]: https://tools.ietf.org/html/rfc8118

File: src/VStore/Sessions/SessionManagementService.cs (hunk 1 above)
Comment (xakep139) [reply src/VStore/Sessions/SessionManagementService.cs]: Also `application/mshelp` (https://bz.apache.org/bugzilla/show_bug.cgi?id=37798) and `application/vnd.ms-htmlhelp` (https://www.iana.org/assignments/media-types/application/vnd.ms-htmlhelp)

File: src/VStore/Sessions/SessionManagementService.cs [hunk 4]
```diff
@@ -52,6 +52,19 @@ public sealed class SessionManagementService
                         { FileFormat.Png, "image/png" }
                 };
 
+        private static readonly IReadOnlyDictionary<string, string> ContentTypesMap =
+            new Dictionary<string, string>
+                {
+                    { FileFormat.Bmp.ToString().ToLowerInvariant(), "image/bmp" },
```
Comment (xakep139) [inline src/VStore/Sessions/SessionManagementService.cs]: https://tools.ietf.org/html/rfc7903

File: src/VStore/Sessions/SessionManagementService.cs [hunk 5]
```diff
@@ -52,6 +52,19 @@ public sealed class SessionManagementService
...
+                    { FileFormat.Gif.ToString().ToLowerInvariant(), "image/gif" },
+                    { FileFormat.Jpeg.ToString().ToLowerInvariant(), "image/jpeg" },
+                    { FileFormat.Jpg.ToString().ToLowerInvariant(), "image/jpeg" },
+                    { FileFormat.Png.ToString().ToLowerInvariant(), "image/png" },
```
Comment (xakep139) [inline src/VStore/Sessions/SessionManagementService.cs]: https://www.iana.org/assignments/media-types/image/png

File: src/VStore/Sessions/SessionManagementService.cs [hunk 6]
```diff
@@ -52,6 +52,19 @@ public sealed class SessionManagementService
                         { FileFormat.Png, "image/png" }
                 };
 
+        private static readonly IReadOnlyDictionary<string, string> ContentTypesMap =
+            new Dictionary<string, string>
+                {
+                    { FileFormat.Bmp.ToString().ToLowerInvariant(), "image/bmp" },
+                    { FileFormat.Gif.ToString().ToLowerInvariant(), "image/gif" },
+                    { FileFormat.Jpeg.ToString().ToLowerInvariant(), "image/jpeg" },
+                    { FileFormat.Jpg.ToString().ToLowerInvariant(), "image/jpeg" },
```
Comment (xakep139) [inline src/VStore/Sessions/SessionManagementService.cs]: https://tools.ietf.org/html/rfc2046

File: src/VStore/Sessions/SessionManagementService.cs [hunk 7]
```diff
@@ -42,7 +42,7 @@ public sealed class SessionManagementService
     {
         private const string PdfHeader = "%PDF-";
 
-        private static readonly Dictionary<FileFormat, string> BitmapImageFormatsMap =
+        private static readonly IReadOnlyDictionary<FileFormat, string> BitmapImageFormatsMap =
```
Comment (denisivan0v) [inline src/VStore/Sessions/SessionManagementService.cs]: It's better to use the file info comes from `ImageSharp` lib to set content type when storing in Ceph. See comments below.

File: src/VStore/Sessions/SessionManagementService.cs [hunk 8]
```diff
@@ -257,6 +270,7 @@ public sealed class SessionManagementService
                     var fileKey = Path.ChangeExtension(uploadSession.SessionId.AsS3ObjectKey(uploadResponse.ETag), fileExtension);
                     var copyRequest = new CopyObjectRequest
                         {
+                            ContentType = ContentTypesMap[fileExtension],
```
Comment (denisivan0v) [inline src/VStore/Sessions/SessionManagementService.cs]: The `EnsureFileContentIsValid` method can return file content type based on it's content (with `ImageSharp` for images, and with static map for other file types) that can be used to set `ContentType` property.

With this we can avoid storing info in `ContentTypesMap` for images.

File: src/VStore/Sessions/SessionManagementService.cs (hunk 7 above)
Comment (xakep139) [reply src/VStore/Sessions/SessionManagementService.cs]: Done in https://github.com/2gis/nuclear-vstore/pull/120/commits/b226aa431e9167fd6ba0b968f10c7b3592b2188c

File: src/VStore/Sessions/SessionManagementService.cs [hunk 9]
```diff
@@ -257,6 +266,7 @@ public sealed class SessionManagementService
                     var fileKey = Path.ChangeExtension(uploadSession.SessionId.AsS3ObjectKey(uploadResponse.ETag), fileExtension);
                     var copyRequest = new CopyObjectRequest
                         {
+                            ContentType = contentType,
```
Comment (xakep139) [inline src/VStore/Sessions/SessionManagementService.cs]: Main change

**Pull request merged**